Xamarin.Forms.Core/ListView.cs

```diff
@@ -437,7 +437,7 @@ public void NotifyRowTapped(int groupIndex, int inGroupIndex, Cell cell = null)
 			_previousGroupSelected = groupIndex;
 
 			// A11y: Keyboards and screen readers can deselect items, allowing -1 to be possible
-			if (cell == null && inGroupIndex != -1)
+			if (cell == null && inGroupIndex >= 0)
```

Comment:
I catch the exception because inGroupIndex was -2. I think it is better to check valid range for index here

Reply:
#8039